test/datatype.test.js

```diff
@@ -206,7 +206,7 @@ describe('datatypes', function() {
 
     it('should set missing optional properties to null', function(done) {
       var EXPECTED = {desc: null, stars: null};
-      TestModel.create({name: 'a-test-name'}, function(err, created) {
+      TestModel.create({}, function(err, created) {
```

Comment:
This test checks what happens when you have `persistUndefinedAsNull` configured and create a model instance that does not have values for all properties. 

Your conclusion seems reasonable to me. I guess my only concern is whether it's ok to pass an empty object to `create`. Perhaps we should add the `name` property to the model definition instead?